run-build

```diff
@@ -173,6 +173,28 @@ class ImageBuildRoot(object):
         for path in to_mount:
             self.mount(path)
 
+        # Remove the tmpfiles.d snippet that sets up symlinks for changes we have
+        # just reversed, and for the dpkg database we are about to move back
+        os.unlink(os.path.join(self.builddir, 'usr/lib/tmpfiles.d/ostree.conf'))
+
+        # Create any files that would normally be handled at runtime. This is
+        # particularly important for /var since it's likely that it was removed
+        # from the ostree commit. Run this inside the chroot to avoid depending on
+        # systemd-tmpfiles on the host system.
+        self._run(['systemd-tmpfiles', '--create', '--remove', '--boot', '-E'])
+
+        # Move the dpkg database back to its normal location
+        os.rename(
+            os.path.join(self.builddir, 'usr/share/dpkg/database'),
+            os.path.join(self.builddir, 'var/lib/dpkg'),
+        )
+
+        # update-catalog requires /var/lib/sgml-base to exist
+        os.makedirs(os.path.join(self.builddir, 'var/lib/sgml-base'))
+
+        # update-xmlcatalog requires /var/lib/xml-core to exist
+        os.makedirs(os.path.join(self.builddir, 'var/lib/xml-core'))
```

Comment:
The comment in `run-build` seems prescient:

```
        # Pull the ostree. Ideally the eosminbase ostree would be used to
        # minimize the buildroot, but that's not released and would fail with
        # --use-production-ostree.
```

I haven't checked but I would be interested to know if these packages are part of eosminbase. If not then these steps would not be necessary.

Reply:
Actually I have a vague memory of seeing errors from `sgml-base` in my toolboxes.

Reply:
Are the errors fatal? Ideally those would have tmpfiles snippets so we didn't need to do anything like this. Or the scripts would, you know, attempt to create the directories instead of assuming they exist? But meh, this is fine.

Reply:
Yeah, unfortunately they are fatal, and a trigger failing is in turn fatal to the `apt-get install` invocation.

I agree that ideally they would have tmpfiles snippets – though I think the Way of the Future in OSTree is to put it in `/usr/share/factory` or whatever it's called?